DataDog/datadog-operator
# [Orch] Add CRD RC handler

### What does this PR do?

Adds remote config support for custom resources in the operator. This new code will merge the user defined configuration with the configuration stored in remote config. Operator users with remote config enabled should no longer require a custom operator configuration to send custom resources.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Minimum Agent Versions

Are there minimum versions of the Datadog Agent and/or Cluster Agent required?

* Cluster Agent: v7.42

### Describe your test plan

- RC is only available on datad0g.com.  Update operator to use staging api key and app key. Update `DD_SITE` to `datad0g.com`
- Make sure operator has list and watch permission on * / *
- Update DatadogAgent to include
```
    orchestratorExplorer:
      enabled: true
    remoteConfiguration:
      enabled: true
```
- Turn on/off custom resources indexing in UI
- You should be able to see this section in `status` of DatadogAgent
```
  remoteConfigConfiguration:
    features:
      orchestratorExplorer:
        customResources:
        - datadoghq.com/v2alpha1/datadogagents
        - mutations.gatekeeper.sh/v1/assign
        - kueue.x-k8s.io/v1beta1/admissionchecks
        - test.com/v1/josh
        ........
```


### Checklist
